keyboards/keychron/c1/keyboard.json

```diff
@@ -0,0 +1,259 @@
+                {"matrix":[4, 8], "x":8.25, "y":4.25},
+                {"matrix":[4, 9], "x":9.25, "y":4.25},
+                {"matrix":[4,10], "x":10.25, "y":4.25},
+                {"matrix":[4,11], "x":11.25, "y":4.25},
+                {"matrix":[4,13], "x":12.25, "y":4.25, "w":2.75},
+                {"matrix":[4,15], "x":16.25, "y":4.25},
+
+                {"matrix":[5, 0], "x":0, "y":5.25, "w":1.25},
+                {"matrix":[5, 1], "x":1.25, "y":5.25, "w":1.25},
+                {"matrix":[5, 2], "x":2.5, "y":5.25, "w":1.25},
+                {"matrix":[5, 6], "x":3.75, "y":5.25, "w":6.25},
+                {"matrix":[5,10], "x":10, "y":5.25, "w":1.25},
+                {"matrix":[5,11], "x":11.25, "y":5.25, "w":1.25},
+                {"matrix":[5,12], "x":12.5, "y":5.25, "w":1.25},
+                {"matrix":[5,13], "x":13.75, "y":5.25, "w":1.25},
+                {"matrix":[5,14], "x":15.25, "y":5.25},
+                {"matrix":[5,15], "x":16.25, "y":5.25},
+                {"matrix":[4,14], "x":17.25, "y":5.25}
+            ]
+        }
+    },
+    "rgb_matrix": {
+        "driver": "sn32f24xb",
+        "speed": 96,
+        "animations": {
+            "breathing": true,
+            "band_spiral_val": true,
+            "cycle_all": true,
+            "cycle_left_right": true,
+            "cycle_up_down": true,
+            "rainbow_moving_chevron": true,
+            "cycle_out_in": true,
+            "cycle_out_in_dual": true,
+            "cycle_pinwheel": true,
+            "cycle_spiral": true,
+            "dual_beacon": true,
+            "rainbow_beacon": true,
+            "jellybean_raindrops": true,
+            "pixel_rain": true,
+            "typing_heatmap": true,
+            "digital_rain": true,
+            "solid_reactive_simple": true,
+            "solid_reactive_multiwide": true,
+            "solid_reactive_multinexus": true,
+            "splash": true,
+            "solid_splash": true
+        },
+        "center_point": [81, 26],
+        "layout": [
+            {"matrix":[0, 0],  "flags":4, "x":0,   "y":0},
+            {"matrix":[0, 1],  "flags":4, "x":12,  "y":0},
+            {"matrix":[0, 2],  "flags":4, "x":25,  "y":0},
+            {"matrix":[0, 3],  "flags":4, "x":37,  "y":0},
+            {"matrix":[0, 4],  "flags":4, "x":50,  "y":0},
+            {"matrix":[0, 5],  "flags":4, "x":62,  "y":0},
+            {"matrix":[0, 6],  "flags":4, "x":75,  "y":0},
+            {"matrix":[0, 7],  "flags":4, "x":87,  "y":0},
+            {"matrix":[0, 8],  "flags":4, "x":100, "y":0},
+            {"matrix":[0, 9],  "flags":4, "x":112, "y":0},
+            {"matrix":[0, 10], "flags":4, "x":124, "y":0},
+            {"matrix":[0, 11], "flags":4, "x":137, "y":0},
+            {"matrix":[0, 12], "flags":4, "x":149, "y":0},
+            {"matrix":[0, 14], "flags":4, "x":147, "y":0},
+            {"matrix":[0, 15], "flags":4, "x":187, "y":0},
+            {"matrix":[0, 16], "flags":4, "x":199, "y":0},
```

Comment:
You seem to have a max `"x":199`, however x_max should be `224` to not need to hack the center value and have effects scale properly. See https://github.com/SonixQMK/qmk_firmware/blob/sn32_develop/docs/features/rgb_matrix.md#common-configuration-common-configuration

Normalize the coordinates for the full range, then the effects will scale properly

Reply:
I have manually recalculated all of the positions.